modules/editor/init.zsh

```diff
@@ -278,6 +278,31 @@ else
   bindkey -M vicmd "/" history-incremental-search-forward
 fi
 
+# Unbound keys in vicmd and viins mode will cause really odd things to happen
+# such as the casing of all the characters you have typed changing or other
+# undefined things. Bind them to the same mode that we are already in so this
+# doesn't happen.
+local unbound_keys=(
+  "${key_info[F1]}"
+  "${key_info[F2]}"
+  "${key_info[F3]}"
+  "${key_info[F4]}"
+  "${key_info[F5]}"
+  "${key_info[F6]}"
+  "${key_info[F7]}"
+  "${key_info[F8]}"
+  "${key_info[F9]}"
+  "${key_info[F10]}"
+  "${key_info[F11]}"
+  "${key_info[F12]}"
+  "${key_info[PageUp]}"
+  "${key_info[PageDown]}"
+)
+for keymap in $unbound_keys; do
+  bindkey -M vicmd "${keymap}" vi-cmd-mode
```

Comment:
Could these be bound to `noop` rather than the vi-mode you're currently in, or is there different behavior when that's done?